web3/contract.py

```diff
@@ -861,6 +859,11 @@ def transact_with_contract_function(contract=None,
             sender = transact_transaction['from']
             nonce = contract.web3.eth.getTransactionCount(sender)
             transact_transaction['nonce'] = nonce
+        if 'gas' not in transact_transaction:
+            gas_estimation_transaction = {k: v for k, v in transact_transaction.items()
+                                          if k not in ('private_key', 'nonce')}
+            gas = contract.web3.eth.estimateGas(gas_estimation_transaction)
+            transact_transaction['gas'] = gas
```

Comment:
For consistency and generality, this estimate can go in `sendRawTransaction()` (similar to the estimate in `sendTransaction()`)

https://github.com/pipermerriam/web3.py/blob/1d551de901d60a08c8cb51170cda44809b67e6af/web3/eth.py#L253